src/serial.rs

```diff
@@ -144,16 +153,41 @@ impl<W: Write> Serial<W> {
     /// # Example
     ///
     /// ```rust
-    /// # use std::io::sink;
+    /// # use std::io::{sink, Error, Result};
+    /// # use std::ops::Deref;
+    /// # use vm_superio::trigger::Trigger;
     /// # use vm_superio::Serial;
     /// # use vmm_sys_util::eventfd::EventFd;
-    /// let intr_evt = EventFd::new(libc::EFD_NONBLOCK).unwrap();
-    /// let serial = Serial::new(intr_evt.try_clone().unwrap(), Vec::new());
     ///
+    /// struct TestEventFd(EventFd);
```

Comment:
I wouldn't name it `Test..` in this context because this shows up in the public documentation. How about `EventFdTrigger`?